### DISCUSSION TITLE
How to student Computer Science as a new UG student?

### DISCUSSION THREAD

**Discussion by AnzhiZhang** [Schweinepriester/github-profile-achievements#672] [Q&A] [answered]
Any links/videos explaining importing?

**Comment by ArielMAJ** [comment, marked as answer]
1. Make sure to go to all your classes;
2. Get to know your professors and what they work on: ask some of them to participate in research projects if you're interested in that;
3. Try to do "relevant" contributions on your GitHub every day, as a way to force you to work on some project every day;
4. Try to think of personal projects in which you use the things you have learned;
5. Try to get friends/people to work on your projects and find people with cool projects for you to work on theirs (basically, make sure to participate in team projects on GitHub);
6. Don't limit yourself to only what you learn in your undergraduate course, make sure to do relevant courses on subjects of your interest: avoid Udemy, do Coursera & edX courses. edX's CS50 courses can give you free certificates outside edX platform. edX can give you 90% discount if you ask them for financial aid and Coursera can give you certificates for free if you ask for financial aid (I've received financial aid in both platforms before).
    1. [CS50x](https://www.edx.org/course/introduction-computer-science-harvardx-cs50x), [CS50P](https://www.edx.org/course/cs50s-introduction-to-programming-with-python?index=product&queryID=1158801f28afa300905d856dcc23417b&position=2) and [other CS50/HarvardX courses](https://www.edx.org/school/harvardx).
    2. [Andrew Ng](https://www.coursera.org/instructor/andrewng) courses if you're interested in AI/ML.
    3. There are many Google/Meta/IBM great courses in [Coursera](https://www.coursera.org/).
    4. If you must do Udemy, I can only recommend [Angela Yu's courses](https://www.udemy.com/user/4b4368a3-b5c8-4529-aa65-2056ec31f37e/) (except the ML one)
 7. Learn [data structures and algorithms](https://www.coursera.org/specializations/data-structures-algorithms) thoroughly before you finish your course.

PS.: Anyone can create a Udemy course, even you and I. Coursera and edX's courses are mostly from big companies and universities. The courses over there feel like taking an actual University course, and they are hard.
